src/System.Private.CoreLib/src/System/Threading/Thread.CoreRT.cs

```diff
@@ -327,14 +327,121 @@ public bool Join(int millisecondsTimeout)
+        {
+            if (!IsYieldProcessorNormalizedInitialized())
+            {
+                InitializeYieldProcessorNormalized();
+            }
+        }
+
+        private static void InitializeYieldProcessorNormalized()
+        {
+            // TODO: critical section so that it only initializes once
+
+            // Intel pre-Skylake processor: measured typically 14-17 cycles per yield
+            // Intel post-Skylake processor: measured typically 125-150 cycles per yield
+            const int DefaultYieldsPerNormalizedYield = 1; // defaults are for when no measurement is done
+            const int DefaultOptimalMaxNormalizedYieldsPerSpinIteration = 64; // tuned for pre-Skylake processors, for post-Skylake it should be 7
+            const int MeasureDurationMs = 10;
+            const int MaxYieldsPerNormalizedYield = 10; // measured typically 8-9 on pre-Skylake
+            const int MinNsPerNormalizedYield = 37; // measured typically 37-46 on post-Skylake
+            const int NsPerOptimialMaxSpinIterationDuration = 272; // approx. 900 cycles, measured 281 on pre-Skylake, 263 on post-Skylake
+            const int NsPerSecond = 1000 * 1000 * 1000;
+
+            ulong ticksPerSecond = HighPerformanceCounter.Frequency;
+            if (ticksPerSecond < 1000 / MeasureDurationMs)
+            {
+                // High precision clock not available or clock resolution is too low, resort to defaults
+                s_yieldsPerNormalizedYield = DefaultYieldsPerNormalizedYield;
+                s_optimalMaxNormalizedYieldsPerSpinIteration = DefaultOptimalMaxNormalizedYieldsPerSpinIteration;
+                return;
+            }
+
+            // Measure the nanosecond delay per yield
+            ulong measureDurationTicks = ticksPerSecond / (1000 / MeasureDurationMs);
+            uint yieldCount = 0;
+            ulong startTicks = HighPerformanceCounter.TickCount;
+            ulong elapsedTicks;
+            do
+            {
+                RuntimeImports.RhSpinWait(10);
+                yieldCount += 10;
+
+                ulong nowTicks = HighPerformanceCounter.TickCount;
+                elapsedTicks = nowTicks - startTicks;
+            } while (elapsedTicks < measureDurationTicks);
+            double nsPerYield = (double)elapsedTicks * NsPerSecond / ((double)yieldCount * ticksPerSecond);
+            if (nsPerYield < 1)
+            {
+                nsPerYield = 1;
+            }
+
+            // Calculate the number of yields required to span the duration of a normalized yield
+            int yieldsPerNormalizedYield = (int)(MinNsPerNormalizedYield / nsPerYield + 0.5);
+            if (yieldsPerNormalizedYield < 1)
+            {
+                yieldsPerNormalizedYield = 1;
+            }
+            else if (yieldsPerNormalizedYield > MaxYieldsPerNormalizedYield)
+            {
+                yieldsPerNormalizedYield = MaxYieldsPerNormalizedYield;
```

Comment:
I don't think this is necessary, as described in the code from CoreCLR the value of `yieldsPerNormalizedYield` here is at most `MinNsPerNormalizedYield` and the value would be closer to what is measured without the extra limit. Probably doesn't matter much though.